builder/azure/arm/step_deploy_template.go

```diff
@@ -71,53 +73,61 @@ func (s *StepDeployTemplate) Cleanup(state multistep.StateBag) {
 	}()
 
 	ui := state.Get("ui").(packersdk.Ui)
-	ui.Say("\nDeleting individual resources ...")
-
 	deploymentName := s.name
 	resourceGroupName := state.Get(constants.ArmResourceGroupName).(string)
-	// Get image disk details before deleting the image; otherwise we won't be able to
-	// delete the disk as the image request will return a 404
-	computeName := state.Get(constants.ArmComputeName).(string)
-	isManagedDisk := state.Get(constants.ArmIsManagedImage).(bool)
-	imageType, imageName, err := s.disk(context.TODO(), resourceGroupName, computeName)
-
-	if err != nil && !strings.Contains(err.Error(), "ResourceNotFound") {
-		ui.Error(fmt.Sprintf("Could not retrieve OS Image details: %s", err))
-	}
-	err = s.delete(context.TODO(), deploymentName, resourceGroupName)
-	if err != nil {
-		s.reportIfError(err, resourceGroupName)
-	}
-
-	// The disk was not found on the VM, this is an error.
-	if imageType == "" && imageName == "" {
-		ui.Error(fmt.Sprintf("Failed to find temporary OS disk on VM.  Please delete manually.\n\n"+
-			"VM Name: %s\n"+
-			"Error: %s", computeName, err))
-		return
-	}
-	if !state.Get(constants.ArmKeepOSDisk).(bool) {
-		ui.Say(fmt.Sprintf(" Deleting -> %s : '%s'", imageType, imageName))
-		err = s.deleteDisk(context.TODO(), imageName, resourceGroupName, isManagedDisk)
+	if s.isVirtualMachineDeployment {
```

Comment:
Since the keyvault case is the guard case I would suggest checking if isVirtualMachineDeployment is false, run the key vault deletion and return earlier. Making the vm deletion the default logic.